tcms/rpc/tests/test_testcase.py

```diff
@@ -603,3 +736,59 @@ def verify_api_with_permission(self):
     def verify_api_without_permission(self):
         with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
             self.rpc_client.TestCase.comments(self.case.pk)
+
+
+class TestAddAttachmentPermissions(APIPermissionsTestCase):
+    permission_label = 'attachments.add_attachment'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.case = TestCaseFactory()
+
+    def verify_api_with_permission(self):
+        self.rpc_client.TestCase.add_attachment(self.case.pk,
+                                                'test.txt',
+                                                'test')
```

Comment:
last parameter should be a base64 encoded string, see below. Otherwise we may be hitting unrelated errors.